docs(link): add good to know comment about scroll behavior (#63152)
vercel/next.js: docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx

<a href="/dashboard#settings">Settings</a>
```

> **Good to know**:
>

> > - If `{children}` from a Layout is not present in the current viewport, Next.js will scroll to the top of the `{children}` instead.

Reviewer: ```suggestion
> - If `{children}` from a Layout is not visible in the viewport, Next.js will scroll to it.
```

> > - If `{children}` from a Layout is not present in the current viewport, Next.js will instead scroll to the top of the `{children}`.

Reviewer: ```suggestion
> - Next.js will scroll to the part of the page that changes as a result of the navigation, if it's not visible in the viewport.
```
